### DISCUSSION TITLE
Nodes turn red when pressing queue prompt - Mac Studio [ PLEASE HELP]

### DISCUSSION THREAD

**Discussion by jaivishal0902** [comfyanonymous/ComfyUI#5558] [Q&A] [answered]
I am a noob and new to ComfyUI. I some how managed to install it on my Mac Studio. I downloaded a json file from an interesting youtube video which i found very useful to my work - https://www.youtube.com/watch?v=Z0xCUZITYZw&t=762s. 

Issue is when i load the test image and queue prompt it doesnt gve me any result sonce few nodes are turning red.
![Screenshot 2024-11-09 at 7 20 00 PM](https://github.com/user-attachments/assets/5506db70-0049-444a-9394-1ae608be3126)
![Screenshot 2024-11-09 at 7 20 29 PM](https://github.com/user-attachments/assets/3e792ecb-226d-43a0-b708-23b1e63fd721)

I would be really thankful for any help.

**Comment by ltdrdata** [comment]
You haven't saved those models in their respective model paths.
First, download the models and save them to the corresponding paths.

You seem to have no prior knowledge about ComfyUI. I recommend watching the following video.
https://www.youtube.com/watch?v=gj6ptjBojl0

**Reply by jaivishal0902** [reply]
Huge thanks. As i said, i am a noob so no clue about anyhting at all. How did i install the ComfyUI on my Mac - hahahaha. One hell of a task it was. Only good thing is, I am an old IT Professional.

Will go through the video and if needed, will bother again. Thanks a lot.

**Comment by jaivishal0902** [comment]
Hello - Your help was quite useful and i managed to fix most of the missing nodes and model issues except this one.
![Screenshot 2024-11-10 at 8 24 55 PM](https://github.com/user-attachments/assets/0f5a183d-4a0c-4a2a-a825-1753475eed4f)
![Screenshot 2024-11-10 at 8 25 28 PM](https://github.com/user-attachments/assets/5264dfed-4d89-4fbb-8046-122566fef0a4)

**Reply by ltdrdata** [reply]
Nope. That node belongs to hd-nodes-comfyui.
![image](https://github.com/user-attachments/assets/28fc279d-9a77-413b-87ac-b6bc9b1e11a8)

You can find it in `Install Missing Custom Nodes`.
![image](https://github.com/user-attachments/assets/45feab3d-97bb-4b61-84f5-5fb28dd944b8)

**Reply by jaivishal0902** [reply]
Thanks again. This is whats happening. Tried fixing it but doesnt work. Am i doing something wrong?

![Screenshot 2024-11-14 at 8 26 18 PM](https://github.com/user-attachments/assets/28590c19-9935-45c2-be81-815dc45aa091)

**Reply by ltdrdata** [reply]
Try fix literally means to "try" fixing. It's not a feature that can fix every possible case.
You need to check the terminal log and inquire about the solution in the corresponding custom node repository.

**Reply by jaivishal0902** [reply]
Hello - Thanks again. I am not too conversant with terminl logs etc. Looks like mission impossible to me sadly.

**Reply by ltdrdata** [reply, marked as answer]
You should be able to find the `comfyui.log` file in the path where you ran ComfyUI.

**Reply by jaivishal0902** [reply]
> You should be able to find the `comfyui.log` file in the path where you ran ComfyUI.

[comfyui.log](https://github.com/user-attachments/files/17965899/comfyui.log)

**Reply by jaivishal0902** [reply]
Attached the log for you to have a look. I cant figure the issue sadly.

**Reply by ltdrdata** [reply]
```
  File "/Users/jaivishal/AI/ComfyUI/nodes.py", line 2022, in load_custom_node
    module_spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 995, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/Users/jaivishal/AI/ComfyUI/custom_nodes/hd_node/__init__.py", line 1, in <module>
    from .HDNodes import NODE_CLASS_MAPPINGS, NODE_DISPLAY_NAME_MAPPINGS
  File "/Users/jaivishal/AI/ComfyUI/custom_nodes/hd_node/HDNodes.py", line 18, in <module>
    from .reactor_swapper import swap_face, get_current_faces_model, analyze_faces, get_face_single
  File "/Users/jaivishal/AI/ComfyUI/custom_nodes/hd_node/reactor_swapper.py", line 11, in <module>
    import insightface
ModuleNotFoundError: No module named 'insightface'
```

That is a typical insightface installation problem.
https://github.com/Gourieff/comfyui-reactor-node#installation
